src/videojs-hls.js

```diff
@@ -1008,17 +999,25 @@ videojs.HlsHandler.prototype.loadSegment = function(segmentInfo) {
     // decrease in network performance or a server issue.
     timeout: (segment.duration * 1.5) * 1000
   }, function(error, request) {
-    // the segment request is no longer outstanding
-    self.segmentXhr_ = null;
+    // This is a timeout of a previously aborted segment request
+    // so simply ignore it
+    if (!self.segmentXhr_ || request !== self.segmentXhr_) {
+      return;
+    }
 
     // if a segment request times out, we may have better luck with another playlist
     if (request.timedout) {
+      // the segment request is no longer outstanding
+      self.segmentXhr_ = null;
```

Comment:
This was something left over when I was befuddled by the randomness of failures. 

I knew something was causing stuff to run out of order. I also knew that the fill/drain cycle is locked by the truthyness of `pendingSegment_` and `segmentXhr_` so I was making sure that either `pendingSegment_` or `segmentXhr_` was always set by not clearing `segmentXhr_` until *after* `pendingSegment_` was set which required re-organization.